openpdf/src/main/java/com/lowagie/text/pdf/PdfPageLabels.java

```diff
@@ -267,16 +267,17 @@ PdfDictionary getDictionary(PdfWriter writer) {
         if (labels == null) {
             return null;
         }
-        Map<PdfObject, PdfObject> numberTree = PdfNumberTree.readTree(labels);
-        Integer[] numbers = new Integer[numberTree.size()];
+        Map<PdfNumber, PdfObject> numberTree = PdfNumberTree.readTree(labels);
+        System.out.println("PageLabelFormats");
+        PdfNumber[] numbers = new PdfNumber[numberTree.size()];
         numbers = numberTree.keySet().toArray(numbers);
         Arrays.sort(numbers);
         PdfPageLabelFormat[] formats = new PdfPageLabelFormat[numberTree.size()];
         String prefix;
         int numberStyle;
         int pagecount;
         for (int k = 0; k < numbers.length; ++k) {
-            Integer key = numbers[k];
+            PdfNumber key = numbers[k];
```

Comment:
Do not see any reason to replace usual Java types with PDF wrapper types if polymorphism is not required. However, you can replace this with primitive `int`

Reply:
Because the keys of the numberTree map are of PdfNumber type.

https://lgtm.com/projects/g/andreasrosdal/OpenPDF/alerts/?mode=list&severity=error

Reply:
Oh, jzz. However, it still looks better to not to use `PdfNumber` but `int` instead and `Integer` as map key by mentioned reason - map keys will be used just as numbers, their according PDF representation is not used, so this is just `int`

Reply:
Using integer here would be a too big change and probably introduce some breaking changes. Don't you think?

Reply:
Right in the next code change you will see `PdfNumber::intValue` invocation - this value is passed to `PdfPageLabel` constructor which accepts `int`

Reply:
Done